js/ui/overlay/z_index.js

```diff
@@ -0,0 +1,31 @@
+import { ensureDefined } from "../../core/utils/common";
+
+let baseZIndex = 1500;
+let zIndexStack = [];
+
+export const base = (ZIndex) => {
+    baseZIndex = ensureDefined(ZIndex, baseZIndex);
+    return baseZIndex;
+};
+
+export const create = (baseIndex = baseZIndex) => {
+    const length = zIndexStack.length;
+    const index = (length ? zIndexStack[length - 1] : baseIndex) + 1;
+
+    zIndexStack.push(index);
+
+    return index;
+};
+
+export const remove = (zIndex) => {
```

Comment:
The method could be simplified with one line.
```suggestion
zIndexStack.splice(zIndexStack.indexOf(zIndex), 1);
```